kerastuner/distribute/oracle_chief.py

```diff
@@ -71,10 +74,20 @@ def start_server(oracle):
     port = os.environ['KERASTUNER_ORACLE_PORT']
     server = grpc.server(
         futures.ThreadPoolExecutor(max_workers=1))
+    oracle_servicer = OracleServicer(oracle)
     service_pb2_grpc.add_OracleServicer_to_server(
-        OracleServicer(oracle), server)
+        oracle_servicer, server)
     server.add_insecure_port('{}:{}'.format(ip_addr, port))
     server.start()
     while True:
         # The server does not block otherwise.
-        time.sleep(10)
+        time.sleep(30)
+
+        if oracle_servicer.stop_triggered:
+            while oracle.ongoing_trials:
+                print(f'Stop is triggered. Remaining open trials: {oracle.ongoing_trials}.')
+                time.sleep(10)
+
+            print('Exiting in 10s.')
+            server.stop(10)
+            break
```

Comment:
Should we raise an error after stopping the Oracle server?

The reason is, we expect users to write their code like:

```
tuner = Tuner(...). # Runs an Oracle server if KERASTUNER_TUNER_ID='chief' environment variable is set
tuner.search(...)
```

Because of this, if the Oracle server stops, the tuner will try and run `tuner.search` but it won't work

Maybe we should raise a `StopIterationError` to prevent this? It will have the unfortunate side effect of having the chief oracle job marked as failure though, so i'm not sure. Wdyt?

Reply:
When the tuner tries tuner.search it, if `overwritten` is not set it will finish searching right away.  That is what I see when I tried it.  
Raising an error may indeed be safer, though?